src/cli/self_update/shell.rs

```diff
@@ -255,6 +256,61 @@ impl UnixShell for Fish {
     }
 }
 
+struct Nu;
+
+impl UnixShell for Nu {
+    fn does_exist(&self, process: &Process) -> bool {
+        // nu has to either be the shell or be callable for nu setup.
+        matches!(process.var("SHELL"), Ok(sh) if sh.contains("nu"))
+            || utils::find_cmd(&["nu"], process).is_some()
+    }
+
+    fn rcfiles(&self, process: &Process) -> Vec<PathBuf> {
+        let p0: Vec<_> = process
+            .var("XDG_CONFIG_HOME")
+            .ok()
+            .iter()
+            .flat_map(|p| {
+                let path = PathBuf::from(p).join("nushell/");
+                [path.join("env.nu"), path.join("config.nu")]
+            })
+            .collect();
+
+        let p1: Vec<_> = process
+            .home_dir()
+            .iter()
+            .flat_map(|p| {
+                let path = PathBuf::from(p).join(".config/nushell/");
+                [path.join("env.nu"), path.join("config.nu")]
+            })
+            .collect();
+
+        [p0, p1].concat()
```

Comment:
Sorry for the back and forth, but one more thing: the `.flat_map()` thing still looks a bit too clever.

Simple procedural code will read even better:

```rs
let mut paths = vec![];

if let Ok(path) = process.var("XDG_CONFIG_HOME") {
  let path = PathBuf::from(path).join("nushell/");
  paths.push(path.join("env.nu"));
  paths.push(path.join("config.nu"));
}

if let Some(path) = process.home_dir() {
  let path = PathBuf::from(path).join("nushell/");
  paths.push(path.join("env.nu"))
  paths.push(path.join("config.nu"));
}

paths